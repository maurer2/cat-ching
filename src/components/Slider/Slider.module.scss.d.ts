import globalClassNames, { ClassNames as GlobalClassNames } from '../..style.d';
declare const classNames: typeof globalClassNames & {
  readonly slider: 'slider';
  readonly slide: 'slide';
};
export default classNames;
export type ClassNames = 'slider' | 'slide' | GlobalClassNames;
