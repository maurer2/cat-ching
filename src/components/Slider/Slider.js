import React from 'react';
import style from './Slider.module.scss';

function Slider(props) {
  return (
    <ul className={style.slider}>
      {props.children.map((child, index) => (
        <li className={style.slide} key={index}>
          {child}
        </li>
      ))}
    </ul>
  );
}

export default Slider;
