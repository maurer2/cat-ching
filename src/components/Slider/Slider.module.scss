.slider {
  display: flex;
  flex-grow: 1;
  margin: 0;
  padding: 0;
  list-style: none;
  overflow-y: scroll;
  scroll-snap-type: x mandatory;
}

.slide {
  display: flex;
  min-width: 100%;
  justify-content: center;
  align-items: center;
  scroll-snap-align: center;

  &:nth-child(odd) {
    background: whitesmoke;
  }
}
