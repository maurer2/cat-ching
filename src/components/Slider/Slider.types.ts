export type SliderProps = Record<string, never>;
