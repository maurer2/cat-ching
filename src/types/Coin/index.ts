export type { CoinData, Coin } from './Coin';
