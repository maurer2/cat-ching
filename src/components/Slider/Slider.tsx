import React, { useRef } from 'react';

import style from './Slider.module.scss';
import * as Types from './Slider.types';

function Slider({ children }: Types.SliderProps): JSX.Element {
  const scrollWrapperElement = useRef<HTMLUListElement | null>(null);

  // function handleSegmentScrolling(event: WheelEvent<HTMLUListElement>) {
  //   const { deltaY, currentTarget } = event;
  //   const slideWidthBB = currentTarget.getBoundingClientRect();
  //   const startScrollPosition = scrollWrapperElement.current?.scrollLeft ?? 0;

  //   // scroll right
  //   if (Math.sign(deltaY) === 1) {
  //     scrollWrapperElement.current?.scrollTo({
  //       left: startScrollPosition + slideWidthBB.width / 2,
  //       behavior: 'smooth',
  //     });
  //   }

  //   // scroll left
  //   if (Math.sign(deltaY) === -1) {
  //     scrollWrapperElement.current?.scrollTo({
  //       left: startScrollPosition - slideWidthBB.width / 2,
  //       behavior: 'smooth',
  //     });
  //   }
  // }

  return (
    <ul
      className={style.slider}
      ref={scrollWrapperElement}
      // onWheel={handleSegmentScrolling}
      data-testid="slider"
    >
      {React.Children.map(children, (child) => (
        <li className={style.slide}>{child}</li>
      ))}
    </ul>
  );
}

export default Slider;
