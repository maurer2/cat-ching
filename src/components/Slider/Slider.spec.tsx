import React from 'react';
import { render } from '@testing-library/react';

import Component from './Slider';
import * as Types from './Slider.types';

describe('AppContainer', () => {
  const propsDefault: Types.SliderProps = {
    children: <div>Children</div>,
  };

  const setup = (props = {}) => render(<Component {...propsDefault} {...props} />);

  it('renders', () => {
    const { queryByTestId } = setup();

    expect(queryByTestId('slider')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('has child elements', () => {
    const { getByText } = setup();

    expect(getByText('Children')).toBeInTheDocument();
  });
});
