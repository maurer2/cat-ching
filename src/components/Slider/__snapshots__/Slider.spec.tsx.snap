// Vitest Snapshot v1

exports[`AppContainer > matches snapshot 1`] = `
<div>
  <ul
    class="_slider_xf5j5_1"
    data-testid="slider"
  >
    <li
      class="_slide_xf5j5_1"
    >
      <div>
        Children
      </div>
    </li>
  </ul>
</div>
`;
